Source/AliveLibAE/MainMenu.cpp

```diff
@@ -3066,33 +3114,25 @@ void MainMenuController::HandleMainMenuUpdate()
             return;
         }
 
-        // Todo: change all field_10_fn_update function types to unsigned.
-        const unsigned int pageUpdateReturnedCam = (this->*(pPage->field_10_fn_update))(inputHeld);
+        const MainMenuNextCam pageUpdateReturnedCam = (this->*(pPage->field_10_fn_update))(inputHeld);
 
-        if (pageUpdateReturnedCam <= 0 || (pageUpdateReturnedCam & 0xFF) == static_cast<unsigned int>(gMap_5C3030.field_4_current_camera))
+        if (pageUpdateReturnedCam.page_update_camera == MainMenuCams::eNoChange || 
+            pageUpdateReturnedCam.page_update_camera == gMap_5C3030.field_4_current_camera)
         {
             // stay on the same screen
             return;
         }
 
-        field_218_target_page_index = static_cast<short>(GetPageIndexFromCam_4D05A0(pageUpdateReturnedCam & 0xFF));
-
-        // The return variable of page update seems to have multiple bits of data masked.
-        auto v19 = (pageUpdateReturnedCam >> 16) & 0xFF;
-        field_21A_target_cam = static_cast<short>(v19);
-        if (v19 == 0xFF)
-        {
-            field_21A_target_cam = -1;
-        }
-
-        isScreenTransitionNecessary = BYTE1(pageUpdateReturnedCam);
+        field_218_target_page_index = static_cast<short>(GetPageIndexFromCam_4D05A0(pageUpdateReturnedCam.page_update_camera));
+        field_21A_target_button_index = pageUpdateReturnedCam.button_idx_to_highlight;
+        isScreenTransitionNecessary = BYTE1(pageUpdateReturnedCam.page_update_camera);
```

Comment:
why is BYTE1 needed? can't this be split into 2 1 byte fields?

Reply:
Good point, though reading the value of `page_update_camera` isn't even needed, since the HIBYTE/BYTE1 of it will always be 0 (as MainMenuCams values only go up to 31).
As a result, `isScreenTransitionNecessary` can always be set to 0.